gravatar/src/main/java/com/gravatar/services/ProfileService.kt

```diff
@@ -71,7 +78,64 @@ public class ProfileService(okHttpClient: OkHttpClient? = null) {
      * @param username The username to fetch the profile for
      * @return The fetched profiles
      */
-    public suspend fun fetchByUsername(username: String): Result<Profile, ErrorType> {
+    public suspend fun fetchByUsername(username: String): Result<LegacyProfile, ErrorType> {
         return fetch(username)
     }
+
+    /**
+     * Fetches a Gravatar profile for the given hash or username.
+     *
+     * @param hashOrUsername The hash or username to fetch the profile for
+     * @return The fetched profile
+     */
+    public suspend fun retrieve(hashOrUsername: String): Result<Profile, ErrorType> = runCatchingService {
```

Comment:
If we are about to change the names how do you feel about using [this naming convention](https://kotlinlang.org/docs/api-guidelines-consistency.html#choose-the-appropriate-error-handling-mechanism:~:text=Consider%20providing%20overloads,methods%20for%20channels.)?
>Consider providing overloads of functions, where one throws an exception, while the other wraps it in a result type instead. In these cases, use the Catching suffix to indicate that exceptions are caught in the function. For example, the standard library has the [run](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run.html) and [runCatching](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run-catching.html) functions using this convention, and the coroutines library has [receive](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-receive-channel/receive.html) and [receiveCatching](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-receive-channel/receive-catching.html) methods for channels.

If not now, something to consider in the future.

Reply:
I like the idea! I've renamed the `retrieve` methods to `retrieveCatching` in order to use the naming convention. I don't want to complicate this PR, so I would create a ticket to add the other methods later. Wdyt?